cpp/test/Slice/errorDetection/Undefined.err

```diff
@@ -1,4 +1,5 @@
-Undefined.ice:5: 'Foo' is not defined
+Undefined.ice:5: 'Fake' is not defined
 Undefined.ice:6: 'Foo' is not defined
-Undefined.ice:7: 'Foo' is not defined
-Undefined.ice:8: 'Foo' is not defined
+Undefined.ice:6: 'Bar' is not defined
```

Comment:
In this line:
`dictionary<Foo, Bar> FooBarDictionary;`
Both `Foo` and `Bar` are undefined, but only `Foo` was reported.
Since the parser skipped ahead to the `;` recovery point, ignoring everything in between.

Now we correctly emit an error for both `Foo` and `Bar` being undefined.